### DISCUSSION TITLE
Invalidation

### DISCUSSION THREAD

**Discussion by tomitrescak** [Asjas/prisma-redis-middleware#137] [Q&A] [answered]
Hello, thank you for a super useful package!

Would you be so kind as to elaborate how the invalidation works?
When model A is updated, does this purge every cached query on model A, or is there a more refined approach.

Also, if a cached query holds "included" nested table data, will these be invalidated as well?
thanks!

**Comment by Asjas** [comment, marked as answer]
>When model A is updated, does this purge every cached query on model A

Yes, that is correct 🙂

>Also, if a cached query holds "included" nested table data, will these be invalidated as well?

[No, at the moment it doesn't.](https://github.com/Asjas/prisma-redis-middleware/issues/105#issue-1206106261) I'm not sure what type of solution would work yet.
